VideoLocker/src/main/java/org/edx/mobile/view/UserProfileFragment.java

```diff
@@ -168,26 +171,25 @@ private void setAccount(@Nullable final Account account) {
                 viewHolder.locationContainer.setVisibility(View.GONE);
             } else {
                 viewHolder.limitedView.setVisibility(View.GONE);
-                if (account.getLanguageProficiencies().isEmpty()) {
-                    viewHolder.languageContainer.setVisibility(View.GONE);
-                } else {
-                    viewHolder.languageContainer.setVisibility(View.VISIBLE);
-                    viewHolder.languageText.setText(
-                            new Locale.Builder()
-                                    .setLanguage(account.getLanguageProficiencies().get(0).getCode())
-                                    .build()
-                                    .getDisplayName());
+                viewHolder.languageContainer.setVisibility(View.GONE);
+                if (!account.getLanguageProficiencies().isEmpty()) {
+                    try {
+                        viewHolder.languageText.setText(
+                                LocaleUtils.getLanguageNameFromCode(account.getLanguageProficiencies().get(0).getCode()));
+                        viewHolder.languageContainer.setVisibility(View.VISIBLE);
+                    } catch (InvalidLocaleException e) {
+                        logger.error(e, true);
```

Comment:
If we can't parse the provided language or region, then I guess it should be treated the same as not provided at all, and the view hidden?